bundles/org.openhab.ui/web/src/assets/definitions/blockly/blocks-notifications.js

```diff
@@ -89,12 +90,417 @@ export default function defineOHBlocks_Notifications (f7, isGraalJs) {
+    let media = javascriptGenerator.valueToCode(block, 'media', javascriptGenerator.ORDER_ATOMIC)
+    let action = javascriptGenerator.valueToCode(block, 'action', javascriptGenerator.ORDER_ATOMIC)
+    const actionType = (action !== '') ? blockGetCheckedInputType(block, 'action') : undefined
+    let actionButton1 = javascriptGenerator.valueToCode(block, 'actionButton1', javascriptGenerator.ORDER_ATOMIC)
+    const actionButton1Type = (actionButton1 !== '') ? blockGetCheckedInputType(block, 'actionButton1') : undefined
+    let actionButton2 = javascriptGenerator.valueToCode(block, 'actionButton2', javascriptGenerator.ORDER_ATOMIC)
+    const actionButton2Type = (actionButton2 !== '') ? blockGetCheckedInputType(block, 'actionButton2') : undefined
+    let actionButton3 = javascriptGenerator.valueToCode(block, 'actionButton3', javascriptGenerator.ORDER_ATOMIC)
+    const actionButton3Type = (actionButton3 !== '') ? blockGetCheckedInputType(block, 'actionButton3') : undefined
+    let users = javascriptGenerator.valueToCode(block, 'usersInput', javascriptGenerator.ORDER_ATOMIC)
+
+    function getActionButtonCode (titleAndAction, buttonActionType) {
+      let actionButtonCode
+      if (buttonActionType === 'String') {
+        titleAndAction = titleAndAction.replace(/'/g, '').split('=')
+        if (titleAndAction.length === 2) {
+          let title = titleAndAction[0]
+          let buttonAction = titleAndAction[1].replace(/ /g, '')
+          actionButtonCode = `.addActionButton('${title}','${buttonAction}')`
+        }
+      } else {
+        actionButtonCode = titleAndAction
+      }
+      return actionButtonCode
+    }
+
+    let iconCode = (icon === '') ? '' : `.withIcon(${icon})`
+    let tagCode = (tag === '') ? '' : `.withTag(${tag})`
+    let headerTitleCode = (headerTitle === '') ? '' : `.withTitle(${headerTitle})`
+    let referenceCode = (reference === '') ? '' : `.withReferenceId(${reference})`
+    let mediaCode = (media === '') ? '' : `.withMediaAttachmentUrl(${media})`
+    let actionCode = (action === '') ? '' : (actionType === 'String') ? `.withOnClickAction(${action})` : `.withOnClickAction('${action}')`
+    let usersCode = (users === '') ? '' : `.addUserId(...${users}.split(','))`
+
+    let actionButton1Code = getActionButtonCode(actionButton1, actionButton1Type)
+    let actionButton2Code = getActionButtonCode(actionButton2, actionButton2Type)
+    let actionButton3Code = getActionButtonCode(actionButton3, actionButton3Type)
+    let logOnlyCode = (type === 'cloud') ? '' : '.logOnly()'
+
+    if (isGraalJs) {
+      return `actions.notificationBuilder(${message})${usersCode}${headerTitleCode}${iconCode}${tagCode}${referenceCode}${mediaCode}${actionCode}${actionButton1Code}${actionButton2Code}${actionButton3Code}${logOnlyCode}.send()\n`
+    } else {
+      throw new Error(unavailMsg)
+    }
+  }
+
+  Blockly.Blocks['oh_cloudNotificationButton'] = {
```

Comment:
Probably better naming:

```suggestion
  Blockly.Blocks['oh_cloudNotificationActionButton'] = {
```

Reply:
If not really necessary I would rather leave it as it is because it "destroys" all my visual test...